cw/capture.py

```diff
@@ -1144,10 +1144,23 @@ def capture_otbn_vertical(ot, ktp, fw_bin, pll_frequency, capture_cfg, device_cf
 
     # Seed the RNG and generate a random fixed seed for all traces of the
     # keygen operation.
-    seed_fixed = ktp.next_key()
-    print(f'fixed seed = {seed_fixed.hex()}')
-    if len(seed_fixed) != seed_bytes:
-        raise ValueError(f'Fixed seed length is {len(seed_fixed)}, expected {seed_bytes}')
+
+    C = ktp.next_key()
+    if len(C) != seed_bytes:
+        raise ValueError(f'Fixed seed length is {len(C)}, expected {seed_bytes}')
+
+    if capture_cfg["test_type"] == 'KEY':
+        ktp.key_len = key_bytes
+        key_fixed = ktp.next_key()
+        if len(key_fixed) != key_bytes:
+            raise ValueError(f'Fixed seed length is {len(key_fixed)}, expected {key_bytes}')
+        ktp.key_len = seed_bytes
+
+        intseed = int.from_bytes(C, byteorder='little') + \
+            int.from_bytes(key_fixed, byteorder='little')
+        seed_fixed = intseed.to_bytes(seed_bytes, byteorder='little')
```

Comment:
What about the XOR with a random seed1?
And shouldn't this be repeated for every trace with a random seed1?

Reply:
The xoring is done by ibex in fuction `p256_ecdsa_gen_secret_key`. In the current binary, ibex receives `seed` and `mask`, and then computes `share0 = seed xor mask` and `share1 = mask`.

Reply:
Yes correct, but I thought we want to pre-calculate the mask (seed2) value from the fixed key and a random (for each trace) seed1, and then pass seed1 and seed2 (mask) to the otbn keygen. (similar thought as in the comment below)

Reply:
hmmm to be honest, I don't have a strong opinion on where to calculate things, but in either case this should be documented in both (python and c) source files.

Reply:
I've now added a comment to explain what's happening on ibex